openmetadata-ui/src/main/resources/ui/src/components/Entity/EntityLineage/CustomNode.utils.tsx

```diff
@@ -67,31 +68,47 @@ export const getColumnHandle = (
   }
 };
 
-export const getExpandHandle = (
-  direction: LineageDirection,
-  onClickHandler: () => void
-) => {
+const ExpandHandle = ({
+  direction,
+  onClickHandler,
+}: {
+  direction: LineageDirection;
+  onClickHandler: (depth: number) => void;
+}) => {
+  const [showExpandAll, setShowExpandAll] = useState(false);
+
   return (
-    <Button
+    <div
       className={classNames(
-        'absolute lineage-node-handle flex-center',
+        'absolute lineage-node-handle-expand-all flex-center',
         direction === LineageDirection.Downstream
           ? 'react-flow__handle-right'
           : 'react-flow__handle-left'
       )}
-      icon={
-        <PlusIcon className="lineage-expand-icon" data-testid="plus-icon" />
-      }
-      shape="circle"
-      size="small"
-      onClick={(e) => {
-        e.stopPropagation();
-        onClickHandler();
-      }}
-    />
+      onClick={(e) => e.stopPropagation()}
+      onMouseOut={() => setShowExpandAll(false)}
+      onMouseOver={() => setShowExpandAll(true)}>
```

Comment:
Can we create callback function and wrap them to useCallback to avoid being creating on each re-renders?